src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -844,6 +846,21 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
     /*package*/ GitClient createClient(TaskListener listener, EnvVars environment, Job project, Node n, FilePath ws) throws IOException, InterruptedException {
 
         String gitExe = getGitExe(n, listener);
+
+        UnsupportedCommand unsupportedCommand = new UnsupportedCommand();
+        for (GitSCMExtension ext : extensions) {
+            ext.determineSupportForJGit(this, unsupportedCommand);
+        }
+
+        listener.getLogger().println("Using Performance Improvement");
+        GitToolChooser estimator = null;
+        for (UserRemoteConfig uc : getUserRemoteConfigs()) {
+            String url = getParameterString(uc.getUrl(), environment);
+            estimator = new GitToolChooser(url, unsupportedCommand.determineSupportForJGit());
+        }
```

Comment:
I'm not following this loop. You will get only the `choose` for the last remote (`estimator` will be overridden in every iteration).

NIT: rename the variable `estimator` to `chooser` for clarity

Reply:
I'm actually always confused about multiple remote configurations. AFAIK, the git plugin supports to build a single unique repo url for a project? Why do we provide a list of user remote configs then?

I've written this part assuming that we won't have a user remote config with a different remote url. Even if we have two or three, all of them would have the same repo url.